dash/resources.py

```diff
@@ -14,48 +14,73 @@ def __init__(self, resource_name):
     def append_resource(self, resource):
         self._resources.append(resource)
 
+    # pylint: disable=too-many-branches
     def _filter_resources(self, all_resources, dev_bundles=False):
         filtered_resources = []
         for s in all_resources:
             filtered_resource = {}
             if 'dynamic' in s:
                 filtered_resource['dynamic'] = s['dynamic']
+            if 'async' in s:
+                if 'dynamic' in s:
+                    raise exceptions.ResourceException(
+                        '{} can\'t have both \'dynamic\' and '
+                        '\'async\'.'.format(json.dumps(filtered_resource))
+                    )
+
+                # Async assigns a value dynamically to 'dynamic'
+                # based on the value of 'async' and config.eager_loading
+                #
+                # True -> dynamic if the server is not eager, False otherwise
+                # 'lazy' -> always dynamic
+                # 'eager' -> dynamic if server is not eager
+                # (to prevent ever loading it)
+                if s['async'] is True:
+                    filtered_resource[
+                        'dynamic'
+                    ] = not self.config.eager_loading
+                else:
+                    filtered_resource['dynamic'] = bool(
+                        (s['async'] == 'eager' and
+                            not self.config.eager_loading)
+                        or s['async'] == 'lazy'
+                    )
```

Comment:
The `# pylint: disable=too-many-branches` might be telling us a subroutine would be useful 😅 
Also your `bool` is redudant.

Reply:
https://github.com/plotly/dash/pull/899/commits/91771d1d94ce69a6033f4a7445dbebd6faa3275f